File: src/python/scraper/videos/fresh.py
from .base.video_scraper import VideoScraper
from .base.parser import parse_response

class FreshScraper(VideoScraper):
    def fresh(self, page: int = 1) -> str:
        if page == 1:
            endpoint = "fresh"
        else:
            endpoint = 'fresh_p'
        data = self.scrape(endpoint, {'page': page - 1} if page else None)
        return parse_response(page, data, self.scrape, is_fresh=True)

File: src/api/videos/fresh.ts
import VideoScraper from "api/videoScraper";
import Parser from "api/utils/parser";

class FreshScraper extends VideoScraper {
  private parser: Parser;
  private endpoint: string;
  constructor(isFresh: boolean = true) {
    super();
    this.parser = new Parser(isFresh);
    this.endpoint = "fresh_p";
  }
  async fresh(
    page: number,
  ): Promise<{ videos: Video[]; pagination: Pagination }> {
    if (page == 0) {
      this.endpoint = "fresh";
    }
    const data = await this.scrape(this.endpoint, { page });
    return this.parser.parseResponse(page, data);
  }
}

export default FreshScraper;